docs/CMakeLists.txt

```diff
@@ -56,6 +52,16 @@ set(DOXYGEN_WARN_IF_UNDOC_ENUM_VAL YES CACHE STRING "")
 set(DOXYGEN_WARN_AS_ERROR YES CACHE STRING "")
```

Comment:
I agree this is not a good default value. I would prefer to be defaulted to NO but having value YES in the CI. I think it would require to update how we generate the doxyfile as these variables are not in cmake cache.

Reply:
`DOXYGEN_WARN_AS_ERROR` is in the cache, since you specified it when calling `set`.

Reply:
True I did not pay attention.

I know that this option may be useful to check the correctness of the documentation. If for example we make a typo in the name of a variable. I remember testing an option that should control if the generation process should stop on the first error or accumulate them but could not make it work at that time.

I would keep it that way for now.